### PR TITLE
GH-818: As a user I need support for dots "." in JavaScript file names

### PR DESCRIPTION
**Pull request by mor-n4**
See #818.

### CODE REVIEW

**Comment by mor-n4** [comment]
Pending:
1. search for `'.'` etc.
2. add test with import from other project.

File: plugins/org.eclipse.n4js.ui/src/org/eclipse/n4js/ui/wizard/workspace/WorkspaceWizardValidatorUtils.java [hunk 1]
```diff
@@ -20,7 +20,7 @@
 public class WorkspaceWizardValidatorUtils {
 
 	private static final Pattern VALID_FOLDER_NAME_PATTERN = Pattern
-			.compile("[a-zA-z_](([a-zA-z_0-9-])|[a-zA-z_0-9-])*");
+			.compile("[a-zA-Z_.0-9-]+");
```
Comment (szarnekow) [inline plugins/org.eclipse.n4js.ui/src/org/eclipse/n4js/ui/wizard/workspace/WorkspaceWizardValidatorUtils.java]: I think '.' is a wildcard in this pattern. I'm not 100% certain, though.

Comment (jpilgrim) [reply plugins/org.eclipse.n4js.ui/src/org/eclipse/n4js/ui/wizard/workspace/WorkspaceWizardValidatorUtils.java]: To save you guys some time (in order to add documentation):

```
Pattern p = Pattern.compile("[a-zA-Z_.0-9-]+");
p.matcher("abc.def").matches();  ==> true
p.matcher("abc!def").matches(); ==> false
```
So, it is not a wildcard here.

**Comment by jpilgrim** [comment]
@mor-n4, I just stumbled over a comment at org.eclipse.n4js.ui.contentassist.N4JSProposalProvider.getProposalFactory(String, ContentAssistContext) concerning names with a dot. Maybe it is worth to have a look there as well.

**Comment by jpilgrim** [comment]
@mor-n4 Some UI tests timed out... Have a look at http://projn4js.ful.informatik.haw-hamburg.de:8080/job/N4JSPR/job/n4js/view/change-requests/job/PR-1324/1/console

```
[INFO] org.eclipse.n4js.ui.tests .......................... FAILURE [  9.307 s]
...
[ERROR] Failed to execute goal org.eclipse.xtend:xtend-maven-plugin:2.19.0:compile (default) on project org.eclipse.n4js.ui.tests: Error compiling xtend sources in '/var/lib/jenkins/workspace/N4JSPR_n4js_PR-1324/tests/org.eclipse.n4js.ui.tests/src'. -> [Help 1]
[ERROR] 
```

**Comment by mor-n4** [comment]
Thanks for the feedback but this PR wasn't ready for review, yet.

The problem is that `.` is still used in several places as a delimiter of FQNs:
* in the FQN property of the reflection,
* in JSDoc links,
* ...

If we allow `.` in folder/module names, all those legacy FQNs break. Changing the format of those FQNs now will require changes in the runtime (e.g. dependency injection) and in client code (stdlib, NFON, ...).

**Comment by mor-n4** [comment]
FYI: the ticket will remain open after merging this; I will check if more spec updates, etc. are useful and provide a second PR if necessary.

**Comment by mor-n4** [comment]
@jpilgrim

> @mor-n4, I just stumbled over a comment at org.eclipse.n4js.ui.contentassist.N4JSProposalProvider.getProposalFactory(String, ContentAssistContext) concerning names with a dot. Maybe it is worth to have a look there as well.

I have created `GH-1546` for that.

**Comment by mor-n4** [comment]
Thanks! Merging ...

**Pull request merged**